modular_skyrat/modules/engineering_override/code/engineering_override.dm

```diff
@@ -0,0 +1,137 @@
+// This module sets airlocks in certain areas to be able to have an Engineer Override on orange alert.
+// Crew with ID cards with the engineering flag will be able to access these areas during those times.
+
+/area
+	/// Is this area eligible for engineer override?
+	var/engineering_override_eligible = FALSE
+
+/**
+ * Set the areas that will receive expanded access for the engineers on an orange alert
+ * Maintenance, bridge, departmental lobbies and inner rooms. No access to security.
+ * Sensitive areas like the vault, command quarters, heads' offices, etc. are not applicable.
+ */
+/area/station/ai_monitored/command/storage/eva
+	engineering_override_eligible = TRUE
+
+/area/station/cargo
+	engineering_override_eligible = TRUE
+
+/area/station/command/bridge
+	engineering_override_eligible = TRUE
+
+/area/station/command/teleporter
+	engineering_override_eligible = TRUE
+
+/area/station/command/gateway
+	engineering_override_eligible = TRUE
+
+/area/station/construction/storage_wing
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/atmos
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/atmospherics_engine
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/storage/tcomms
+	engineering_override_eligible = TRUE
+
+/area/station/hallway/secondary/service
+	engineering_override_eligible = TRUE
+
+/area/station/maintenance
+	engineering_override_eligible = TRUE
+
+/area/station/medical
+	engineering_override_eligible = TRUE
+
+/area/station/service
+	engineering_override_eligible = TRUE
+
+/area/station/science
+	engineering_override_eligible = TRUE
+
+/obj/machinery/door
+	/// Determines if engineers get access to this door on orange alert
+	var/engineering_override = FALSE
+
+/// Check for the three states of open access. Emergency, Unrestricted, and Engineering Override
+/obj/machinery/door/allowed(mob/Mob)
+	if(emergency)
+		return TRUE
+
+	if(unrestricted_side(Mob))
+		return TRUE
+
+	if(engineering_override)
+		var/mob/living/carbon/human/user = Mob
+		var/obj/item/card/id/card = user.get_idcard(TRUE)
+		if(istype(user))
+			if(ACCESS_ENGINEERING in card.access)
+				return TRUE
+	return ..()
+
+/// Activate the airlock overrides, called by the change of alert level
+/proc/enable_engineering_access()
+	for(var/area/station_area in get_areas(/area/station))
+		if(!station_area.engineering_override_eligible)
+			continue
+
+		for(var/obj/machinery/door/airlock/airlock in station_area)
+			airlock.engineering_override = TRUE
+			airlock.normalspeed = FALSE
```

Comment:
Having an airlock open for 8 seconds instead of 1.5 means areas will vent faster.

Reply:
Alright, that's fair with me, but you should still probably mention that somewhere, otherwise it's going to be surprising to those that get affected by it.